app/views.py

```diff
@@ -0,0 +1,44 @@
+from flask import session, redirect, url_for, render_template
+from app import app
+from datetime import timedelta
+import pymesync
+import forms
+
+
+# Expires users after 30 minutes OF UNACTIVITY
+@app.before_request
+def make_session_permanent():
+    session.permanent = True
+    app.permanent_session_lifetime = timedelta(minutes=30)
+
+
+@app.route('/')
+def index():
+    return 'Welcome to timesync-frontend.'
+
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    form = forms.LoginForm()
+
+    # TODO: Flash message on invalid submission
+    if form.validate_on_submit():
+        username = form.username.data
+        password = form.password.data
+
+        # Login to TimeSync
+        ts = pymesync.TimeSync(baseurl=app.config['TIMESYNC_URL'])
+        token = ts.authenticate(username=username,
+                                password=password,
+                                auth_type="password")
+
+        # TODO: Better error handling
+        if 'error' in token or 'pymesync error' in token:
+            print token
+            return 'There was an error.', 500
+
+        session['username'] = username
+        session['ts'] = token
```

Comment:
Is it going to be confusing having `ts` be a `pymesync.TimeSync` instance but `session['ts']` a JSON object containing `{'token': 'thereallylongtoken'}`.

You may also want to keep in mind that `session['ts']` does not just contain a token, but as explained above, a JSON object *containing* a token. You will have to keep this in mind for re-auth in other views when you have to pass **just** a token to initiate pymesync.

The alternative to that is to set `session['ts']` (or `session['token']`) to `token['token']` so you don't have to worry about it later. You already verified there is no `error` in `token` so you shouldn't see any key errors. If you want to be even more explicit though you could do something like:

```
if 'token' in token:
    session['token'] = token['token']
else:
    print "unexpected response " + token